core/src/main/scala/org/apache/spark/storage/DiskBlockManager.scala

```diff
@@ -52,6 +53,14 @@ private[spark] class DiskBlockManager(conf: SparkConf, deleteFilesOnStop: Boolea
   // of subDirs(i) is protected by the lock of subDirs(i)
   private val subDirs = Array.fill(localDirs.length)(new Array[File](subDirsPerLocalDir))
 
+  /**
+   * Create merge directories
+   */
+  createLocalDirsForMergedShuffleBlocks(conf)
+
+  private[spark] lazy val activeMergedShuffleDirs: Option[Array[File]] =
+    findActiveMergedShuffleDirs(conf)
+
```

Comment:
Why not populate this as result of `createLocalDirsForMergedShuffleBlocks` ?

Reply:
If this executor is not creating the merged directory, then currently it seems that activeMergedShuffleDirs will be empty. Check line 216.

Reply:
BTW this is no longer needed. The dirs are passed to the getMerged... APIs.